code/modules/mob/living/simple_animal/hostile/terror_spiders/terror_spiders.dm

```diff
@@ -0,0 +1,826 @@
+						targets2 += H
+					else
+						// they are far away, and either we're not very aggressive, or they are dying already
+						// either way, not much point in targeting them
+						// if they shoot us, of course, then we will consider them a valid target
+				else
+					if (H.can_inject(null,0,"chest",0))
+						targets1 += H
+					else if (H in enemies)
+						targets2 += H
+					else
+						targets3 += H
+					// first, go after targets we can inject - lightly armored humans.
+					// second, go after humans that attacked us.
+					// third, go after other humans.
+			else if (istype(H, /mob/living/simple_animal))
+				var/mob/living/simple_animal/hostile/poison/terror_spider/M = H
+				if (M.force_threshold > melee_damage_upper)
+					// If it has such high armor it can ignore any attack we make on it, ignore it.
+				else if (M in enemies)
+					targets2 += M
+				else
+					targets3 += M
+			else
+				if (H in enemies)
+					targets2 += H
+				else
+					targets3 += H
+		for(var/obj/mecha/M in mechas_list)
+			if(get_dist(M, src) <= vision_range && can_see(src, M, vision_range))
+				if(get_dist(M, src) <= 2)
+					targets2 += M
+				else
+					targets3 += M
+		if (health < maxHealth)
+			// very unlikely that we're being shot at by a space pod - so only check for this if our health is lower than max.
+			for(var/obj/spacepod/S in spacepods_list)
+				if(get_dist(S, src) <= vision_range && can_see(src, S, vision_range))
+					targets3 += S
+			// Turret-targeting code is broken. Also such a rare case. We leave it commented for now.
+			/*
+			for(var/obj/machinery/porta_turret/R in view(src, vision_range))
+				if (!R.stat)
+					// spiders will target turrets that shoot them, by default
+					targets3 += R
+			*/
+		if (targets1.len)
+			return targets1
+		else if (targets2.len)
+			return targets2
+		else
+			return targets3
+	else if (ai_type == 1)
+		// DEFEND SELF ONLY
+		//var/list/Mobs = hearers(vision_range, src) - src
+		for(var/mob/living/H in view(src, vision_range))
+		//for(var/mob/H in Mobs)
+			if (H.stat == 2)
+				// dead mobs are ALWAYS ignored.
+			else if (!stat_attack && H.stat == 1)
+				// unconscious mobs are ignored unless spider has stat_attack
+			else if (H in enemies)
+				targets1 += H
+		for(var/obj/mecha/M in mechas_list)
+			if(M in enemies && get_dist(M, src) <= vision_range && can_see(src, M, vision_range))
+				targets1 += M
+		for(var/obj/spacepod/S in spacepods_list)
+			if(S in enemies && get_dist(S, src) <= vision_range && can_see(src, S, vision_range))
+				targets1 += S
+		return targets1
+	else if (ai_type == 2)
+		// COMPLETELY PASSIVE
+		return list()
+
+/mob/living/simple_animal/hostile/poison/terror_spider/LoseTarget()
+	if (target && isliving(target))
+		var/mob/living/T = target
+		if (T.stat > 0)
+			killcount++
+			regen_points += regen_points_per_kill
+	attackstep = 0
+	attackcycles = 0
+	..()
+
+/mob/living/simple_animal/hostile/poison/terror_spider/AttackingTarget()
```

Comment:
This has been refactored in latest commit to eliminate all the istype(src, specific_spider) checks, and about half of the conditionals. Class-specific specials are now handled by spider_specialattack, a proc that gets overridden by subtypes.